torch/distributed/_tensor/device_mesh.py

```diff
@@ -169,15 +170,17 @@ def _init_process_groups(self):
                     f"has mesh {other_mesh}!"
                 )
 
-        # groups created by dimension, each dimension should have exact
-        # one valid process group per rank
-        dim_groups: List[ProcessGroup] = []
+        # group ranks/tags associated with each mesh dimension, each mesh dimension should
+        # have one sub-group per rank
+        dim_group_ranks: List[List[int]] = []
+        dim_group_tags: List[str] = []
```

Comment:
suggestion: 
since we're mostly (if not always) using tag and list of ranks at the same time, shall we combine (zip together) them into a single `List[Tuple[str, List[int]]]` where each element in `List` is a `Tuple` of a tag and a subgroup?

Reply:
sure, updated